src/mesh/bih_tree.cc

```diff
@@ -41,6 +41,7 @@ BIHTree::~BIHTree() {
 
 
 void BIHTree::add_boxes(const std::vector<BoundingBox> &boxes) {
+	main_box_ = BoundingBox( boxes[0].min() ); // fix problem with undefined main_box_

```

Comment:
Add explicit constructor call for main_box_ in the BIH_Tree constructor.
Initialize only if elements_ are empty.

Modify BoundingBox default constructor to initialize to NaNs ... in the parallel DH branch.